app/src/pages/login/LoginForm.tsx

```diff
@@ -16,25 +13,32 @@ type LoginFormParams = {
 export function LoginForm() {
   const navigate = useNavigate();
   const [error, setError] = useState<string | null>(null);
-  const [commit, isCommitting] = useMutation<LoginFormMutation>(graphql`
-    mutation LoginFormMutation($email: String!, $password: String!) {
-      login(input: { email: $email, password: $password })
-    }
-  `);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
   const onSubmit = useCallback(
-    (params: LoginFormParams) => {
-      setError(null);
-      commit({
-        variables: params,
-        onCompleted: () => {
-          navigate("/");
-        },
-        onError: () => {
-          setError("Invalid Login");
-        },
-      });
+    async (params: LoginFormParams) => {
+      setError(() => null);
+      setIsLoading(() => true);
```

Comment:
what's the rationale for the callback syntax here?

Reply:
it's not needed, i can revert

Reply:
i thought this was safer in general, is that not the case?